File: scripts/outbound-call.js
// Trigger an outbound admission call.
//   node scripts/outbound-call.js +9198XXXXXXXX "Student Name"
import { placeCall } from "../src/twilio/outbound.js";

const to = process.argv[2];
const name = process.argv[3] || "";

if (!to) {
  console.error('Usage: node scripts/outbound-call.js <+E164 number> ["Student Name"]');
  process.exit(1);
}

try {
  const sid = await placeCall(to, name);
  console.log(`Calling ${to} (${name || "unknown"}) — Call SID: ${sid}`);
} catch (err) {
  console.error(err.message);
  process.exit(1);
}

File: scripts/outbound_call.py
"""Trigger an outbound admission call.

  python scripts/outbound_call.py +9198XXXXXXXX "Student Name"
"""
import asyncio
import sys

sys.path.insert(0, __import__("os").path.dirname(__import__("os").path.dirname(__import__("os").path.abspath(__file__))))

from app.twilio_outbound import place_call  # noqa: E402


async def main():
    if len(sys.argv) < 2:
        print('Usage: python scripts/outbound_call.py <+E164 number> ["Student Name"]')
        sys.exit(1)
    to = sys.argv[1]
    name = sys.argv[2] if len(sys.argv) > 2 else ""
    try:
        sid = await place_call(to, name)
        print(f"Calling {to} ({name or 'unknown'}) — Call SID: {sid}")
    except Exception as e:
        print(str(e))
        sys.exit(1)


if __name__ == "__main__":
    asyncio.run(main())
